test/unit/ui.js

```diff
@@ -87,38 +132,206 @@ describe('ui class', () => {
         assert.equal(null, ui.components.product[0].config.node);
         ui.destroyComponent('product', ui.components.product[0].model.id);
       });
     });
+
+    it('binds escape key on iframe if it exists', () => {
+      productConfig.iframe = true;
+      const escSpy = sinon.stub(UI.prototype, '_bindEsc');
+      ui = new UI(client);
+
+      return ui.createComponent('product', productConfig).then(() => {
+        assert.calledOnce(escSpy);
+        escSpy.restore();
+        ui.components.product = []; // parentNode does not exist on product.node so .destroyComponent does not work here
+      });
+    });
+
+    it('catches any error from component.init and notifies errorReporter', () => {
+      initStub.restore(); // removes the regular initStub to replace with the initStub that throws an error
+      const integrations = {errorReporter: {notifyException: sinon.spy()}};
+      const errorInitStub = sinon.stub(Product.prototype, 'init').returns(Promise.reject({errors: [{message: 'rejected.'}]}));
+      ui = new UI(client, integrations);
+
+      ui.createComponent('product', productConfig).then(() => {
+        assert.throws(ui.createComponent, Error);
+        assert.calledOnce(integrations.errorReporter.notifyException);
+        errorInitStub.restore();
+        ui.destroyComponent('product', ui.components.product[0].model.id);
+      });
+    });
+
+    it('grabs node from _queryEntryNode if no node is passed in from config', () => {
+      const queryEntryNodeSpy = sinon.spy(UI.prototype, '_queryEntryNode');
+      productConfig.node = null;
+
+      return ui.createComponent('product', productConfig).then(() => {
+        assert.calledOnce(queryEntryNodeSpy);
+        queryEntryNodeSpy.restore();
+      });
+    });
   });
 
   describe('destroyComponent', () => {
     it('removes component and calls its destroy method', () => {
       const testCart = {
         model: {
-          id: 123
+          id: 123,
         },
-        destroy: sinon.spy()
-      }
+        destroy: sinon.spy(),
+      };
       ui.components.cart.push(testCart);
       ui.destroyComponent('cart', 123);
       assert.equal(0, ui.components.cart.length);
       assert.calledOnce(testCart.destroy);
     });
   });
+
+  describe('trackComponent', () => {
+    let trackComponentStub;
+
+    beforeEach(() => {
+      trackComponentStub = sinon.stub(Tracker.prototype, 'trackComponent');
+    });
+
+    afterEach(() => {
+      trackComponentStub.restore();
+    });
+
+    it('if the component is productSet, track each product in set', () => {
+      const productSet = {
+        trackingInfo: [1, 2, 3, 4, 5],
+      };
+
+      ui.trackComponent('productSet', productSet);
+      assert.callCount(trackComponentStub, 5);
+    });
+
+    it('calls trackComponent with the given component type', () => {
+      const component = {
+        trackingInfo: 'test',
+      };
+
+      ui.trackComponent('product', component);
```

Comment:
Do you think we need to explicitly test each component type, or would it suffice to just test a single generic component type?